### PR TITLE
Fixing pack basepath use for relative paths and when it's not specified

### PR DESCRIPTION
**Pull request by toddm**
Found issue while testing with dotnet pack.  Added tests to cover the cases.

@joelverhagen

### CODE REVIEW

File: test/NuGet.Clients.Tests/NuGet.CommandLine.Test/NuGetPackCommandTest.cs [hunk 1]
```diff
@@ -3274,6 +3275,189 @@ public void PackCommand_PackJsonCorrectLibPathInNupkgWithOutputName()
...
+                    files,
+                    new string[]
+                    {
+                        @"lib\net46\proj1.dll",
+                    });
+            }
+        }
+
+        [Fact]
+        public void PackCommand_BuildProjectJsonWithRelativeBasePath()
```
Comment (joelverhagen) [inline test/NuGet.Clients.Tests/NuGet.CommandLine.Test/NuGetPackCommandTest.cs:3401]: Can we have a test for relative `-basepath` and different working directory (e.g. `C:\`)?

**Comment by joelverhagen** [comment]
Address comment then :shipit:

File: test/NuGet.Clients.Tests/NuGet.CommandLine.Test/NuGetPackCommandTest.cs (hunk 1 above)
Comment (toddm) [inline test/NuGet.Clients.Tests/NuGet.CommandLine.Test/NuGetPackCommandTest.cs:3401]: PackCommand_BuildProjectJsonWithRelativeBasePath already does what you ask.  The project.json is not in the working directory, and the basepath is relative.

Comment (toddm) [inline test/NuGet.Clients.Tests/NuGet.CommandLine.Test/NuGetPackCommandTest.cs:3401]: Confirmed with Joel that it's ok as is.

**Pull request merged**